download/download_worker.go

```diff
@@ -119,25 +106,48 @@ func (w *DownloadWorker) NewEvent(incoming events.Message) {
 
 // Download the given object from css
 func (w *DownloadWorker) DownloadCSSObject(org string, objType string, objId string, filePath string, nmpName string) error {
-	filePath = path.Join(filePath, nmpName)
 	glog.Infof(dwlog(fmt.Sprintf("Attempting to download css file %v/%v/%v to file %v", org, objType, objId, filePath)))
 	objMeta, err := exchange.GetObject(w, org, objId, objType)
 	if err != nil {
 		return fmt.Errorf("Failed to get metadata for css object %v/%v/%v. Error was: %v", org, objType, objId, err)
 	}
 
-	err = exchange.GetObjectData(w, org, objType, objId, filePath, objId, objMeta, w.client)
-	if err != nil {
-		w.Messages() <- events.NewNMPDownloadCompleteMessage(events.NMP_DOWNLOAD_COMPLETE, exchangecommon.STATUS_DOWNLOAD_FAILED, nmpName, nil, nil)
-		return fmt.Errorf("Failed to get data for object %v/%v/%v. Error was: %v", org, objType, objId, err)
+	filePath = path.Join(filePath, nmpName)
+
+	if (objMeta.ChunkSize != 0 && objMeta.ChunkSize < objMeta.ChunkSize) || int(objMeta.ObjectSize) > w.Config.GetFileSyncServiceMaxDataChunkSize() {
+		offsetStep := w.Config.GetFileSyncServiceMaxDataChunkSize()
+		startOffest := 0
+		if objMeta.ChunkSize != 0 && objMeta.ChunkSize < offsetStep {
```

Comment:
Can you remind me why we need objMeta.ChunkSize for offsetStep? Can we just use the value of ` w.Config.GetFileSyncServiceMaxDataChunkSize()`? I might forget something about this part if I implemented sync service in this way

Reply:
Fixed to use the FSS config variable. I had thought that would sometimes not be set, but there is a default